src/hdmint/wallet.cpp

```diff
@@ -324,22 +315,29 @@ bool CHDMintWallet::SeedToZerocoin(const uint512& seedZerocoin, GroupElement& co
     return true;
 }
 
-CKeyID CHDMintWallet::GetZerocoinSeedID(uint32_t n){
+CKeyID CHDMintWallet::GetZerocoinSeedData(int32_t nCount, uint160 hashSeedMaster){
     // Get CKeyID for n from mintpool
-    std::pair<CKeyID,uint32_t> mintPoolEntry;
-    if(!mintPool.Get(n, mintPoolEntry)){
+    uint256 hashPubcoin;
+    std::pair<uint256,MintPoolEntry> mintPoolEntryPair;
+
+    while(!mintPool.Get(nCount, hashSeedMaster, mintPoolEntryPair)){
```

Comment:
If caller pass invalid hashSeedMaster by accident then this loop will run forever.

Reply:
`hashSeedMaster` will always be set by this point. However I will add a check that it is initialized.

Reply:
it's safe from infinity loop now.